### ISSUE TITLE
Build continues after .bff parse error, causing a segfault

### ISSUE THREAD

**Issue by msg-programs** [fastbuild/fastbuild#979]
Here is the `fbuild.bff` that causes the issue for me. It's invalid, but it shouldn't crash:
```
.Compiler = '/usr/bin/gcc'

ObjectList('objs')
{
    .CompilerOptions    = '-c "%1" -o "%2"'
    .CompilerInputPath  = './'
    .CompilerInputPattern = '*.c' // removing this causes the program to behave as expected (abort, no crash)
}

Executable()
{
    .Linker = '/usr/bin/gcc'
    .LinkerOutput   = './test'
    .LinkerOptions  = '"%1" -o "%2"'
    .Libraries      = {'objs'}
}

Alias('all') {.Targets = 'test'}
```
When putting a file `test.c` (any content) in the same dir as the build file, the following happens:
```
$ fbuild -verbose
BFF file '/.../fbuild.bff' has changed (reparsing will occur).
Loading BFF '/.../fbuild.bff'
/.../fbuild.bff(3,1): FASTBuild Error #1101 - ObjectList() - Missing required property 'CompilerOutputPath'.
ObjectList('objs')
^
\--here
3> -Build: 0 ms /usr/bin/gcc                                   
6> Dir: '/.../|*.c|true||' (found 1 files)
 - /.../test.c
6> -Build: 0 ms /.../|*.c|true||
2> -Build: 0 ms /.../test.c
4> -Build: 0 ms ?AutoCompiler?/usr/bin/gcc                     
Segmentation fault (core dumped)
```
I've tried compiling fbuild from source with debug flags (x64Linux-Debug), which produces the following:
```
[same as above, then:]

-------- ASSERTION FAILED --------
/.../fastbuild/Code/Tools/FBuild/FBuildCore/Graph/NodeGraph.cpp(900): Assert: IsCleanPath( objectName )
-----^^^ ASSERTION FAILED ^^^-----

Illegal instruction (core dumped)
```
Is this issue known? I can try to provide more info if needed.

Running Ubuntu 22.04 64 Bit, with FASTBuild 1.09.

**Comment by ffulin** [comment]
Thanks for reporting this bug.

It looks like the error handling in this case is missing a "return false" after emitting the error:
```
        // .CompilerOutputPath is required when compiling files (not needed if only creating a PCH)
        if ( m_CompilerOutputPath.IsEmpty() )
        {
            Error::Error_1101_MissingProperty( iter, function, AStackString<>( "CompilerOutputPath" ) );
        }
```

**Comment by ffulin** [comment]
This bug has been fixed (https://github.com/fastbuild/fastbuild/commit/5b276019cc9b3e623870d99d718530eb406de544). This fix will be in the next release (v1.10). Thanks for reporting the issue!

I'll keep this issue open until v1.10 is released.

**Comment by ffulin** [comment]
v1.10 has been released and includes a fix for this issue.
